doc/new_boilerplate_migration.md

```diff
@@ -803,14 +803,14 @@ UiFactory<FooProps> createFooHoc(UiFactory otherFactory) {
   Object closureVariable; 
   // ... 
 
-  final FooHoc = uiFunctionComponent<FooProps>(
+  final FooHoc = uiFunction<FooProps>(
```

Comment:
We basically decided:
- it was better to keep it consistent with class component factories
- it avoids triggering the type_annotate_public_apis lint

Reply:
I didn't add any examples of it, but declaring a top level function component without LHS typing does work. So should I change it so that they can't be declared without LHS typing?

Reply:
Oh nice! Hmmm I guess if there's not a reason to disallow it, we shouldn't disallow it.

Are there tests for declaring them without LHS typing?

Reply:
The parsing and codegen tests have with/without LHS typing declarations sprinkled throughout, but I'll add one to the integration tests as well